src/toolcalls/chain/chainToolCallOperation.ts

```diff
@@ -43,7 +47,10 @@ export interface ChainToolCallOperation<T> {
   walletMustMatchChainID?: boolean;
 
   /** Validates the provided parameters match requirements */
-  validate(params: T, walletStore: WalletStore): boolean | Promise<boolean>;
+  validate(
+    params: T,
+    walletConnection: WalletConnection,
```

Comment:
We don't need to pass the whole store in - we are only using the snapshot of the current connection.